subgraphs/isle-finance-v0/protocols/isle-finance-v0/src/mapping.ts

```diff
@@ -0,0 +1,916 @@
+  POOL_CONFIGURATOR_BYTES32 as POOL_CONFIGURATOR_ID,
+  LOAN_MANAGER_BYTES32 as LOAN_MANAGER_ID,
+  WITHDRAWAL_MANAGER_BYTES32 as WITHDRAWAL_MANAGER_ID,
+} from "./constants";
+import { MarketDailySnapshot, Token, _Loan } from "../../../generated/schema";
+import { ERC20, Transfer } from "../../../generated/templates/Pool/ERC20";
+
+/////////////////////
+//// Pool Events ////
+/////////////////////
+
+//
+// Pool-side contracts created event
+export function handleProxyCreated(event: ProxyCreated): void {
+  // create which template depends on the event.params.id
+  const id = event.params.id;
+  if (id === POOL_CONFIGURATOR_ID) {
+    PoolConfiguratorTemplate.create(event.params.proxyAddress);
+  } else if (id === LOAN_MANAGER_ID) {
+    LoanManagerTemplate.create(event.params.proxyAddress);
+  } else if (id === WITHDRAWAL_MANAGER_ID) {
+    WithdrawalManagerTemplate.create(event.params.proxyAddress);
+  }
+}
+
+export function handleInitialized(event: Initialized): void {
+  PoolTemplate.create(event.params.pool_);
+
+  const poolContract = Pool.bind(event.params.pool_);
+  const outputToken = new TokenManager(
+    event.params.pool_,
+    event,
+    TokenType.NON_REBASING
+  );
+
+  const tryInputToken = poolContract.try_asset();
+  if (tryInputToken.reverted) {
+    log.error("[handleInitialized] Pool contract {} does not have an asset", [
+      event.params.pool_.toHexString(),
+    ]);
+    return;
+  }
+
+  const manager = new DataManager(
+    event.params.pool_,
+    tryInputToken.value,
+    event,
+    getProtocolData()
+  );
+  
+  // new market, set both interest rates as 0
+  const market = manager.getMarket();
+  manager.getOrUpdateRate(
+    InterestRateSide.BORROWER,
+    InterestRateType.VARIABLE,
+    BIGDECIMAL_ZERO
+  );
+  manager.getOrUpdateRate(
+    InterestRateSide.LENDER,
+    InterestRateType.VARIABLE,
+    BIGDECIMAL_ZERO
+  );
+
+  const tryConfigurator = poolContract.try_configurator();
+  if (tryConfigurator.reverted) {
+    log.error(
+      "[handleInitialized] Pool contract {} does not have a configurator",
+      [event.params.pool_.toHexString()]
+    );
+    return;
+  }
+
+  // update market with isle specifics
+  market.id = event.params.pool_;
+  market.name = outputToken.getToken().name;
+  market.outputToken = outputToken.getToken().id;
+  market.outputTokenSupply = BIGINT_ZERO;
+  market.outputTokenPriceUSD = BIGDECIMAL_ZERO;
+  market.exchangeRate = BIGDECIMAL_ZERO; // exchange rate = (inputTokenBalance / outputTokenSupply) OR (totalAssets() / totalSupply())
+  market.isActive = false; // controlled with setAsActive
+  market.canBorrowFrom = false; // controlled with setAsActive
+  market.canUseAsCollateral = false; // collateral is posted during loans separate from any deposits
+  market.borrowedToken = tryInputToken.value;
+  market.stableBorrowedTokenBalance = BIGINT_ZERO;
+  market._poolConfigurator = tryConfigurator.value;
+  market.save();
+}
+
+/*
+//
+// handles borrow creations for loans
```

Comment:
Why are these logic commented out?

(If it is for future use, we can keep it in another branch, but I don't think we should push all of this commented code to the main branch.)